### PR TITLE
48 implement the frontend part of login

### LINKED ISSUE #48: Implement the frontend part of login
Probably using https://github.com/jpadilla/ember-simple-auth-token

### PR DESCRIPTION
**Pull request by ebolloff**
This fixes #48. I'll review @eugenk , he'll review @phyrog  and me.

### CODE REVIEW

File: app/controllers/application.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  authenticatedUser: Ember.inject.service()
```
Comment (phyrog) [inline app/controllers/application.js]: It's inferred from the field name.

Comment (eugenk) [inline app/controllers/application.js]: Ah okay. Thanks

File: app/adapters/application.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 import Ember from 'ember';
 import DS from 'ember-data';
 import ENV from "../config/environment";
+import DataAdapterMixin from 'ember-simple-auth/mixins/data-adapter-mixin';
```
Comment (phyrog) [inline app/adapters/application.js]: Agreed.

**Comment by phyrog** [comment]
I just added a new commit that extracts the session handler logic (`signin` and `signout`) to the application controller. Now if we have a component that needs to be able to sign in or out (e.g. a full size login page), we pass the action handlers as parameters to the component. The component has its own action handlers that only concern the component itself (update its view and/or internal state), that in turn call the passed action handlers to actually sign in and out. The `signin` action also gets a callback function that is called with the login-promise, so the component can change to a loading state and out of it once the promise resolves.

TL;DR: Controller passes `signin` action to component, component handles actions in component and calls passed `signin` action with a callback function that receives a promise to update its state in response to the promise.

**Comment by eugenk** [comment]
:+1:

File: app/adapters/application.js (hunk 2 above)
Comment (ebolloff) [inline app/adapters/application.js]: 👍 Done.

**Comment by phyrog** [comment]
I just pushed a new commit, that reloads the complete local data store on login/logout to prevent data leakage.
